packages/common-server/src/parser.ts

```diff
@@ -301,16 +303,51 @@ export class SchemaParserV2 extends ParserBaseV2 {
     });
 
     const addConnections = (parent: SchemaProps) => {
-      _.forEach(parent.children, (ch) => {
-        const child = schemasDict[ch];
-        if (child) {
-          DNodeUtils.addChild(parent, child);
+      _.forEach(parent.children, (childId) => {
+        const child = schemasDict[childId];
 
-          addConnections(child);
+        if (child) {
+          if (child.parent === null) {
+            // Child does not have a parent pointers hence we can linkup the parent child as is.
+            DNodeUtils.addChild(parent, child);
+            addConnections(child);
+          } else {
+            // Child already contains a parent pointer hence we need to create a schema clone
```

Comment:
Hmm, what if child.parent === parent? Do we need to deep clone in that case? 

I'm looking through the surrounding code, not sure I _fully_ understand it w/o utilizing a debugger haha.  But if I have a schema like 

```yml
- id: foo
  children:
    - fooschild
    - wildchild

- id: fooschild
  parent: foo
  
 - id: wildchild
```

Then, at this point in the code
- the node with id `fooschild` has a parent pointer to `foo`?
- 'wildchild' has no parent pointer?
- 'foo' has 2 children?

Reply:
Interesting case. We would end up cloning fooschild in this case (even though we don't really have to). However, it should not hurt us since we will take out the original fooschild reference from the parent. 

Putting some print statements for this case we end up with:
```
Starting to process parent 'foo' with children: 'fooschild,wildchild'
Processing 'fooschild' which has parent 'foo'
Cloning child: 'fooschild'
Starting to process parent 'fooschild_9q31V7vxlhhzzF0odUraW' with children: ''
Parent 'foo' children: 'wildchild,fooschild_9q31V7vxlhhzzF0odUraW'
Processing 'wildchild' which has parent 'null'
Starting to process parent 'wildchild' with children: ''
Parent 'foo' children: 'wildchild,fooschild_9q31V7vxlhhzzF0odUraW'
Starting to process parent 'root' with children: ''
```